src/irr/asset/COBJMeshFileLoader.cpp

```diff
@@ -76,32 +86,87 @@ COBJMeshFileLoader::COBJMeshFileLoader(IAssetManager* _manager) : AssetManager(_
+			(*bindings)[i].binding = i;
+			(*bindings)[i].samplers = samplers;
+		}
+
+
+		auto ds1layout = getDefaultAsset<ICPUDescriptorSetLayout, IAsset::ET_DESCRIPTOR_SET_LAYOUT>("irr/builtin/descriptor_set_layout/basic_view_parameters", AssetManager);
+		core::smart_refctd_ptr<ICPUDescriptorSetLayout> ds3Layout = core::make_smart_refctd_ptr<ICPUDescriptorSetLayout>(bindings->begin(), bindings->end());
+		SPushConstantRange pcRng;
+		pcRng.stageFlags = ICPUSpecializedShader::ESS_FRAGMENT;
+		pcRng.offset = 0u;
+		pcRng.size = sizeof(208u);
+		layout = core::make_smart_refctd_ptr<ICPUPipelineLayout>(&pcRng, &pcRng + 1, nullptr, std::move(ds1layout), nullptr, std::move(ds3Layout));
+
+	}
+	std::pair<core::smart_refctd_ptr<ICPUSpecializedShader>, core::smart_refctd_ptr<ICPUSpecializedShader>> shaders[2];
+	shaders[0] = {
+		std::move(getDefaultAsset<ICPUSpecializedShader, IAsset::ET_SPECIALIZED_SHADER>("irr/builtin/shaders/loaders/mtl/vertex_no_uv.vert", AssetManager)),
+		std::move(getDefaultAsset<ICPUSpecializedShader, IAsset::ET_SPECIALIZED_SHADER>("irr/builtin/shaders/loaders/mtl/fragment_no_uv.frag", AssetManager))
+	};
+	shaders[1] = {
+		std::move(getDefaultAsset<ICPUSpecializedShader, IAsset::ET_SPECIALIZED_SHADER>("irr/builtin/shaders/loaders/mtl/vertex_uv.vert", AssetManager)),
+		std::move(getDefaultAsset<ICPUSpecializedShader, IAsset::ET_SPECIALIZED_SHADER>("irr/builtin/shaders/loaders/mtl/fragment_uv.frag", AssetManager))
+	};
+
+	constexpr size_t DS1_METADATA_ENTRY_CNT = 3ull;
+	core::smart_refctd_dynamic_array<IPipelineMetadata::ShaderInputSemantic> shaderInputsMetadata = core::make_refctd_dynamic_array<decltype(shaderInputsMetadata)>(DS1_METADATA_ENTRY_CNT);
+	{
+		ICPUDescriptorSetLayout* ds1layout = layout->getDescriptorSetLayout(1u);
+
+		constexpr IPipelineMetadata::E_COMMON_SHADER_INPUT types[DS1_METADATA_ENTRY_CNT]{ IPipelineMetadata::ECSI_WORLD_VIEW_PROJ, IPipelineMetadata::ECSI_WORLD_VIEW, IPipelineMetadata::ECSI_WORLD_VIEW_INVERSE_TRANSPOSE };
+		constexpr uint32_t sizes[DS1_METADATA_ENTRY_CNT]{ sizeof(SBasicViewParameters::MVP), sizeof(SBasicViewParameters::MV), sizeof(SBasicViewParameters::NormalMat) };
+		constexpr uint32_t relOffsets[DS1_METADATA_ENTRY_CNT]{ offsetof(SBasicViewParameters,MVP), offsetof(SBasicViewParameters,MV), offsetof(SBasicViewParameters,NormalMat) };
+		for (uint32_t i = 0u; i < DS1_METADATA_ENTRY_CNT; ++i)
+		{
+			auto& semantic = (shaderInputsMetadata->end() - i - 1u)[0];
+			semantic.type = types[i];
+			semantic.descriptorSection.type = IPipelineMetadata::ShaderInput::ET_UNIFORM_BUFFER;
+			semantic.descriptorSection.uniformBufferObject.binding = ds1layout->getBindings().begin()[0].binding;
+			semantic.descriptorSection.uniformBufferObject.set = 1u;
+			semantic.descriptorSection.uniformBufferObject.relByteoffset = relOffsets[i];
+			semantic.descriptorSection.uniformBufferObject.bytesize = sizes[i];
+			semantic.descriptorSection.shaderAccessFlags = ICPUSpecializedShader::ESS_VERTEX;
+		}
+	}
```

Comment:
layouts, shaders and metadata are common for both actual MTL pipelines and a dummy default one.

Also why are you creating this default pipeline in OBJ loader, and not MTL loader?

Reply:
because if there is no mtl file, the pipeline for a mesh buffer is assigned in obj loader

Reply:
the default result of a loader should be handled by the loader.